crates/core/tedge_api/src/workflow/supervisor.rs

```diff
@@ -185,13 +186,16 @@ impl WorkflowSupervisor {
             });
         };
 
-        let version = command_state.workflow_version();
+        let Some(version) = &command_state.workflow_version() else {
+            return Err(WorkflowExecutionError::MissingVersion);
+        };
```

Comment:
I have to revert this change. Indeed, this might be a source of issue if the agent is updated whilst there is a pending operation. If the previous agent was not featuring command versions, then the resumed operation should not be rejected by the new agent: the current version of the workflow must be used.